### ISSUE TITLE
RScan (RChain Block Explorer)

### ISSUE THREAD

**Issue by allancto** [rchain/bounties#978]
### Benefit to RChain

Etherscan is one of the most widely used blockchain related tools. This issue if properly formulated will be sponsored by the dApp development community, as communicated by @kdvalentine. 

TBD: actual project proposal.

### Budget and Objective

_Estimated Budget of Task:_ **$[TBD]**
_Estimated Timeline Required to Complete the Task:_ **[TBD]**
_How will we measure completion?_ **[TBD]**

### Legal

_Task Submitter shall not submit Tasks that will involve RHOC being transacted in any manner that (i) jeopardizes RHOC’s status as a software access token or other relevant and applicable description of the RHOC as an “asset”—not a security— or (2) violates, in any manner, applicable U.S. Securities laws._

**Comment by dckc** [comment]
w.r.t. the obligation of the TAC to decide on proposals within 5 days, let's treat this as not (yet) a task proposal. Many / most issues labelled **Help Wanted** and **needs-SMART-objective** can work this way, especially with **[TBD]** in the Task Identification Form / ISSUE_TEMPLATE.

[Task Approval][1] says "tasks are represented as github issues" but it doesn't say that all github issues are task proposals.

[1]: https://github.com/rchain/bounties/wiki/Task-Approval

**Comment by allancto** [comment]
@dckc agreed completely. The **Help Wanted** and **needs-SMART-objective** indicate this. The issue is posted because it will have sponsorship and we'd like to make that known and encourage contributors with talent and skills in this area to step up with a team and SMART plan to execute.

**Comment by dckc** [comment]
The [BasicWallet.rho](https://github.com/rchain/rchain/blob/dev/casper/src/main/rholang/BasicWallet.rho) contract connects purses (especially REV) purses to public keys, which makes RChain wallets much like Ethereum accounts.

But I wonder about the analog to Ethereum "internal" transactions. An RChain blockchain explorer seems to be complicated by the fact that REV can be in a purse _not_ attached to any public key, but just floating around in the tuplespace. I suppose they're still connected to the REV mint, so maybe finding REV purposes doesn't amount to solving the halting problem.

Also, I wonder what the RChain equivalent of an Ethereum transaction id is. Maybe it's the output of `@"blake2b256Hash"!([nonce, amount, retCh].toByteArray(), *hashOut)`? Odd... that signature isn't over the public key of the destination wallet but over the (typically unforgeable) name of the destination purse.

I should probably post to https://forum.rchain.coop/ about this...

**Comment by allancto** [comment]
@dckc i have two thoughts in this area. One is that I'd like to see (different issue) some forensic tools on etherscan that allow us to answer questions like where alledged funds transfers actually happened and why (recent chatter in telegram), and statistics similar to those provided by @desperado (David Currin) in response to questions about the rev swap,  and frankly for my own personal curiosity about what kind of information others may be able to find about me and you by applying moderately sophisticated search tools. 

With respect to RChain i believe information recorded on the blockchain will be exponentially richer. Contracts in progress (continuations) may speak volumes. LADL based search will provide an incredible tool for understanding how the platform is used. RScan is only a toe in the water, what kinds of information will wait to be gathered from our RChain mainnet and dApps deployed at scale?

**Comment by JoshOrndorff** [comment]
> allow us to answer questions like where alledged funds transfers actually happened and why

I totally agree about where, but I think why is beyond the scope of a block explorer. I like this project idea and will sponsor a reasonable budget for it when it becomes clear how to do that.
